api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -106,6 +107,25 @@ public User getUserByUsername(String username) {
 		return users.get(0);
 	}
 	
+	
+	/**
+	 * @see org.openmrs.api.UserService#getUserByEmail(java.lang.String)
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public User getUserByEmail(String email) {
+		return (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.eq("email", email).ignoreCase()).uniqueResult();	
+	}
+	
+	/**
+	 * @see org.openmrs.api.UserService#getUserByActivationKey(java.lang.String)
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public User getUserByActivationKey(String activationKey) {
+		return (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.like("activationKey", activationKey, MatchMode.START)).uniqueResult();	
```

Comment:
To make this concrete you need to compare the keys for equality before returning the match